### PR TITLE
#1228 run all tests under multiroot

### PR DESCRIPTION
**Pull request by DonJayamanne**
This PR will run all of the old unit tests against a multi-root environment.

FYI - a hack is in place that will need to be removed (details here https://github.com/DonJayamanne/pythonVSCode/issues/1321).

All tests pass, unfortunately some of them fail randomly (running the python 'unit tests' tests).
Will look into fixing all of them with the other PRs (or possibly when I rewrite to use proper DI to get around execution of Python modules).

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -103,9 +103,9 @@ const hygiene = exports.hygiene = (some, options) => {
             .split(/\r\n|\r|\n/)
             .forEach((line, i) => {
                 if (/^\s*$/.test(line)) {
-                    // empty or whitespace lines are OK
+                    // empty or whitespace lines are OK.
```
Comment (brettcannon) [inline gulpfile.js]: If you want to make the comments be properly formatted, capitalization is probably worth it (for the same reasons for adding a period; you know no one accidentally deleted a comment line).

File: src/client/providers/importSortProvider.ts [hunk 2]
```diff
@@ -1,57 +1,57 @@
...
 export class PythonImportSortProvider {
-    public sortImports(extensionDir: string, document: vscode.TextDocument): Promise<vscode.TextEdit[]> {
+    public async sortImports(extensionDir: string, document: vscode.TextDocument): Promise<vscode.TextEdit[]> {
         if (document.lineCount === 1) {
-            return Promise.resolve([]);
+            return [];
+        }
+        // isort does have the ability to read from the process input stream and return the formatted code out of the output stream.
+        // However they don't support returning the diff of the formatted text when reading data from the input stream.
+        // Yes getting text formatted that way avoids having to create a temporary file, however the diffing will have.
```
Comment (brettcannon) [inline src/client/providers/importSortProvider.ts]: I think you have a trailing period that isn't supposed to be there.

File: src/client/providers/importSortProvider.ts [hunk 3]
```diff
@@ -1,57 +1,57 @@
...
+        const args = settings.sortImports.args.join(' ');
+        let isortCmd = '';
+        if (typeof isort === 'string' && isort.length > 0) {
+            if (isort.indexOf(' ') > 0) {
+                isortCmd = `"${isort}" "${filePath}" --diff ${args}`;
+            }
+            else {
+                isortCmd = `${isort} "${filePath}" --diff ${args}`;
+            }
+        } else {
```
Comment (brettcannon) [inline src/client/providers/importSortProvider.ts]: Is the style:
```javascript
}
else {
```
or
```javascript
} else {
```
?

Comment (DonJayamanne) [reply src/client/providers/importSortProvider.ts]: Hmm, the formatter should have picked that up.

Comment (DonJayamanne) [reply src/client/providers/importSortProvider.ts]: fixed.

File: src/client/providers/importSortProvider.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/providers/importSortProvider.ts]: Fixed

File: gulpfile.js (hunk 1 above)
Comment (DonJayamanne) [reply gulpfile.js]: Fixed

**Pull request merged**